Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean

```diff
@@ -100,6 +102,28 @@ lemma isIso_lanAdjunction_counit_app_iff (G : D ⥤ H) :
     IsIso ((L.lanAdjunction H).counit.app G) ↔ G.IsLeftKanExtension (𝟙 (L ⋙ G)) :=
   (isLeftKanExtension_iff_isIso _ (L.lanUnit.app (L ⋙ G)) _ (by simp)).symm
 
+noncomputable section Colim
+
+variable {C : Type u₁} {D : Type u₁} [Category.{v₁} C] [Category.{v₃} D]
+variable {H : Type (max u₁ u₂)} [Category.{max u₁ v₂} H]
+
+/-- Composing the left Kan extension of `L : C ⥤ D` with `colim` on shapes `D` is isomorphic
+to `colim` on shapes `C`. -/
+@[simps!]
+def lanCompColimIso (L : C ⥤ D) [∀ (G : C ⥤ H), L.HasLeftKanExtension G]
+    [HasColimitsOfShape C H] [HasColimitsOfShape D H] : L.lan ⋙ colim ≅ colim (C := H) :=
```

Comment:
These is a more general construction, given `F'` a left Kan extension of `F` along `L`, then any colimit cocone for `F` gives a colimit cocone for `F'`:
```
variable (F' : D ⥤ H) {L : C ⥤ D} {F : C ⥤ H} (α : F ⟶ L ⋙ F') [F'.IsLeftKanExtension α]

@[simps]
noncomputable def coconeOfIsLeftKanExtension (c : Cocone F) : Cocone F' where
  pt := c.pt
  ι := F'.descOfIsLeftKanExtension α _ c.ι

def isColimitCoconeOfIsLeftKanExtension {c : Cocone F} (hc : IsColimit c) :
    IsColimit (F'.coconeOfIsLeftKanExtension α c) where
  desc s := hc.desc (Cocone.mk _ (α ≫ whiskerLeft L s.ι))
  fac s := by
    have : F'.descOfIsLeftKanExtension α ((const D).obj c.pt) c.ι ≫ 
        (Functor.const _).map
          (hc.desc (Cocone.mk _ (α ≫ whiskerLeft L s.ι))) = s.ι :=
      F'.hom_ext_of_isLeftKanExtension α _ _ (by aesop_cat)
    exact congr_app this
  uniq s m hm := hc.hom_ext (fun j ↦ by
    have := hm (L.obj j)
    nth_rw 1 [← F'.descOfIsLeftKanExtension_fac_app α ((const D).obj c.pt)]
    dsimp at this ⊢
    rw [assoc, this, IsColimit.fac, NatTrans.comp_app, whiskerLeft_app])
```
Then, `lanCompColimIso` should be redefined as `NatIso.ofComponents` of the isomorphisms obtained by `coconePointUniqueUpToIso` using suitable colimit cocones.

Reply:
Are you sure the naturality can be proven if we use `coconePointUniqueUpToIso`? I feel like we lack coherence if we don't construct the iso explicitely.

Reply:
(see attempt in the commit below)

Reply:
Ah, nvm, I think it is possible :grin:

Reply:
Good! But note that the general construction (anything but `lanCompColimIso` and its dual) should go in the more basic file `KanExtension.Basic`.